lib/bulbs-elements/store/store.js

```diff
@@ -42,13 +65,15 @@ function dispatchFunction (store, action, payload) {
     console.time(`dispatch ${action.type}`);
   }
 
-  let nextState = cloneObject(store.state);
-  let actionState = nextState[action.fieldName];
-  let actionNextState = cloneObject(actionState);
-  actionNextState = action(actionNextState, payload, store);
-  nextState[action.fieldName] = actionNextState;
+  let priorState = store.state;
+  let nextState = Object.assign({}, priorState);
+  let priorFieldValue = priorState[action.fieldName];
+  let nextFieldValue = cloneObject(priorFieldValue);
```

Comment:
I guess what I meant was should we put more protection around this. I feel like this is an easy way for bugs to slip in since we're providing not-quite immutable structures.